addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1359,4 +1385,16 @@ public void playMusicVoiceCommand(Device device, String providerId, String voice
         String postData = gson.toJson(startRoutineRequest);
         makeRequest("POST", alexaServer + "/api/behaviors/preview", postData, true, true, null);
     }
+
+    public JsonEqualizer getEqualizer(Device device) throws IOException, URISyntaxException {
```

Comment:
This class is huge (over 1000 lines) and you keep adding methods to add. Is it possible to factor out some groups of functionality for fellow developers to easier understand your code?

Also: Every public method **should** have a javadoc explaining what it does. Try to reduce the public interface of a class, and you need to write less documentation.

Reply:
I big refactoring is already planned, but this will be a lot of work and will not be part of this PR. My plan for this is, to create handler for each channel, the handler will encapsulate  the REST call including the Json-classes. This will make the Connection class very small, because all the commands an information functions are moved to the handler and the EchoHandler will also be much more easier because it will only have a list of the CommandHandler.

Reply:
Thank you, I will check it